src/otx/core/data/dataset/keypoint_detection.py

```diff
@@ -55,15 +55,14 @@ def __init__(
 
         self.dm_subset = self._get_single_bbox_dataset(dm_subset)
 
-        if self.dm_subset.categories():
+        # arrow doesn't follow common coco convention, no need to fetch kp-specific labels
+        if self.dm_subset.categories() and data_format != "arrow":
             kp_labels = self.dm_subset.categories()[AnnotationType.points][0].labels
             self.label_info = LabelInfo(
                 label_names=kp_labels,
                 label_groups=[],
                 label_ids=[str(i) for i in range(len(kp_labels))],
             )
-        else:
```

Comment:
I think we should explicitly assign self.label_info in case of 'arrow' format. Overwise -> raise an error, since NullLabelInfo will break the training at the end.

Reply:
By entering this if/else label info (Null or not) was already generated. The culprit is that it is correct by this moment for arrow, but for coco it is not, and therefore we have an extra labels extraction procedure in this constructor

Reply:
The reference implementation (which is used for arrow) is here: https://github.com/openvinotoolkit/training_extensions/blob/3c9c606589a0053471e17810a500d6fb62eebcad/src/otx/core/data/dataset/base.py#L99
If a change in `NullLabelInfo` is required, it should be done there to cover all the tasks